### ISSUE TITLE
Have fontnames() print out the output of pango-list

### ISSUE THREAD

**Issue by rlafuente** [shoebot/shoebot#344]
I was reading about `fontnames()` in the Nodebox-PyObjC docs and thought we have an easy way to support it. 

And it's a useful function to include, instead of sending the user to their terminal.

**Comment by stuaxo** [comment]
Awesome :)

**Comment by karstenw** [comment]
Please make sure, the fontnames() result elements can be used with font(). That was the main purpose.

**Comment by stuaxo** [comment]
That sounds like a good basis for a unit test.

**Comment by rlafuente** [comment]
> Please make sure, the fontnames() result elements can be used with font(). That was the main purpose.

Definitely, will keep this in mind. I don't know how I didn't notice its companion fontfamilies(), will give some thought on how this can work with the Pango way of dealing with fonts.

(as an aside, really happy to see you here @karstenw and grateful for your input!)

**Comment by karstenw** [comment]
> Definitely, will keep this in mind. I don't know how I didn't notice its companion fontfamilies(), will give some thought on how this can work with the Pango way of dealing with fonts.

fontfamilies came to life because OSX offered it easily. I don't use it; but could. fontnames() is from my POV the most important/useful extension I did. 

The most extensions I made are in

https://github.com/karstenw/nodebox-pyobjc/blob/master/nodebox/util/__init__.py

Have a look at imagefiles()

> (as an aside, really happy to see you here @karstenw and grateful for your input!)
I totally hope you guys are successful with shoebot because Nodebox is one of the factors that keeps me on the mac platform... and I'm not as enthusiastic as I used to be.

**Comment by stuaxo** [comment]
TLDR: I do have a plan to address this, but it will have to happen after our next release, which is 1.4.

Longer version:
Our 1.4 release should be really good, apart from new features we have lots more test coverage which keep things fairly robust, our docs are also getting lots of love and we are getting some nice new examples.

Re: success: 
For the next thing (maybe called Shoebot 2.0) I intend to take everything we've learned during Shoebot development (and by gradually understanding more of the Nodebox codebase) and make something explicitly cross platform.

Cairo + Pango are ostensibly cross platform, but are hard for people to install in various environments, the codebase is not in as good a state as Nodebox as well (we were learning as we went, we have accumulated various hacks, at the same time better approaches and new technologies have arrived).

The new thing, will have the backend properly separated out, so on OSX it could use PyObjc, possibly adopting the Nodebox code, on Linux cairo and Pango, and elsewhere I'm not sure.

I have plans quite a bit beyond this, but the first step, is getting a really good final release of shoebot 1.x out, so we have a nice art tool out while we build the future.

**Comment by rlafuente** [comment]
\o/
